packages/vite/src/node/config.ts

```diff
@@ -1081,25 +1104,69 @@ async function loadConfigFromBundledFile(
   }
 }
 
+const orders = ['pre', null, 'post'] as const
+
 async function runConfigHook(
   config: InlineConfig,
-  plugins: Plugin[],
-  configEnv: ConfigEnv
+  configEnv: ConfigEnv,
+  plugins: Plugin[]
 ): Promise<InlineConfig> {
-  let conf = config
-
-  for (const p of getSortedPluginsByHook('config', plugins)) {
-    const hook = p.config
-    const handler = hook && 'handler' in hook ? hook.handler : hook
-    if (handler) {
-      const res = await handler(conf, configEnv)
-      if (res) {
-        conf = mergeConfig(conf, res)
+  const calledPlugins = new Set<Plugin>()
+  const callConfigHandler = async (plugin: Plugin) => {
+    calledPlugins.add(plugin)
+    const handler =
+      plugin.config && 'handler' in plugin.config
+        ? plugin.config.handler!
+        : plugin.config!
+
+    const result = await handler(config, configEnv)
+    if (result) {
+      config = mergeConfig(config, result)
+    }
+  }
+
+  const enforce = (p: Plugin) =>
+    (p.config && ('handler' in p.config ? p.config.order : p.enforce)) || null
```

Comment:
This deviates from current behavior (which uses `getSortedPluginsByHook`) in that a plugin without a `config` handler **object** will be sorted by its `enforce` property, whereas previously `undefined` was used to sort it.

Reply:
I'm not currently maintaining a separate array for ordering based on `plugin.config.order` because that makes insertion of nested plugins more complicated.